src/modules/socket/server/socket_utils.cpp

```diff
@@ -88,10 +88,13 @@ std::optional<ip_addr_t> get_address(const sockaddr_storage& sstorage)
     }
     case AF_INET6: {
         auto sa6 = reinterpret_cast<const sockaddr_in6*>(&sstorage);
+        if (IN6_IS_ADDR_UNSPECIFIED(&sa6->sin6_addr)) {
```

Comment:
I guess this would change it so that IPV6_V6ONLY socket option is always treated as false.  Previously it seems it was always true.

I think it might be better to add a v6only option to this function and only do this when NOT v6only.

Calling functions would get the v6only flag from the PCB something like this:
addr = get_address(storage, IP_IS_V6_VAL(pcb->local_ip));